networks/suzuka/suzuka-full-node/src/manager.rs

```diff
@@ -27,6 +29,25 @@ impl Manager<SuzukaPartialNode<Executor>> {
     }
 
     pub async fn try_run(&self) -> Result<(), anyhow::Error> {
+
+        let (stop_tx, mut stop_rx) = tokio::sync::watch::channel(());
+        tokio::spawn({
+            let mut sigterm = signal(SignalKind::terminate()).context("Can't register to SIGTERM.")?;
+            let mut sigint = signal(SignalKind::interrupt()).context("Can't register to SIGKILL.")?;
+            async move {
+                loop {
+                    tokio::select! {
+                        _ = sigterm.recv() => (),
+                        _ = sigint.recv() => (),
+                    };
```

Comment:
And again, this whole business could be replaced by a future doing a single `select!` because we're not interested in polling for more termination signals after receiving one.

Reply:
Remember that `Signal::recv` is not special in the way low-level `signal` callbacks are: it's called in a normal application thread and already uses some synchronization magic under the hood that's triggered by the actual callbacks installed by Tokio.

Reply:
Yes, that's what I assumed too. I just don't know how to explain why this works reliable and the setup you describe does not yet. So, let's save this for a future PR.